src/porepy/compositional/compositional_mixins.py

```diff
@@ -0,0 +1,1087 @@
+                        )
+                        if self.equilibrium_type is not None
+                        else phase.partial_fraction_of[component](subdomains).value(
+                            self.equation_system, state
+                        )
+                    )
+                    for component in phase
+                ]
+            )
+            for phase in self.fluid_mixture.phases
+        ]
+
+        return FluidState(
+            z=z,
+            y=y,
+            sat=sat,
+            phases=[PhaseState(x=x_) for x_ in x],
+        )
+
+    def _create_fractional_variable(
+        self,
+        name: str,
+        subdomains: list[pp.Grid],
+    ) -> None:
+        """Helper method to create individual variables."""
+        self.equation_system.create_variables(
+            name=name, subdomains=subdomains, tags={"si_units": "-"}
+        )
+
+    def create_variables(self) -> None:
+        """Creates the sets of required variables for a fluid mixture.
+
+        1. :meth:`overall_fraction` is called to assign
+           :attr:`~porepy.compositional.base.Component.fraction` to components.
+        2. :meth:`solute_fraction` is called to assign
+           :attr:`~porepy.compositional.base.Compound.solute_fraction_of` for each solute
+           in a compound.
+        3. :meth:`saturation` is called to assign
+           :attr:`~porepy.compositional.base.Phase.saturation` to phases.
+
+        If a local :attr:`equilibrium_type` is defined, it introduces additionally
+
+        4. :attr:`~porepy.compositional.base.Phase.fraction` to phases by calling
+           :meth:`phase_fraction'
+        5. :attr:`~porepy.compositional.base.Phase.extended_fraction_of` for each phase and
+           component by calling :meth:`extended_fraction`
+
+        If a local equilibrium is defined :meth:`partial_fractions` returns dependent
+        operators by normalizing extended fractions. Otherwise partial fractions are
+        also introduced as as independent operators by calling :meth:`partial_fraction`
+        for each phase and component in that phase, and assigning it to
+        :attr:`~porepy.compositional.base.Phase.partial_fraction_of`.
+
+        """
+        if not hasattr(self, "fluid_mixture"):
+            raise CompositionalModellingError(
+                "Cannot create fluid mixture variables before defining a fluid mixture."
+            )
+
+        rcomp = self.fluid_mixture.reference_component
+        rphase = self.fluid_mixture.reference_phase
+        subdomains = self.mdg.subdomains()
+
+        # NOTE: The creation of variables seems repetative (it is), but it is done this
+        # way to preserve a certain order (component-wise, phase-wise and familiy-wise
+        # for each family of fractions)
+
+        ## Creation of feed fractions
+        for component in self.fluid_mixture.components:
+            if component != rcomp:  # will be called last
```

Comment:
Because the feed fraction of the reference component can be eliminated by unity.
For this the other feed fractions must already be established.
Added that to the comment.